src/librustc_codegen_llvm/builder.rs

```diff
@@ -564,6 +564,14 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
 
 
     pub fn range_metadata(&self, load: ValueRef, range: Range<u128>) {
+        if self.sess().target.target.arch == "amdgpu" {
```

Comment:
Basically use a check like this instead of `ignore_inline_never`. Both are LLVM bugs AFAICT.